### PR TITLE
Updates list of built-in IO transforms

### PR DESCRIPTION
**Pull request by jkff**
Some new ones have been added, some in-progress ones have been finished, some aren't being worked on (JSON).

### CODE REVIEW

**Comment by melap** [comment]
retest this please

**Comment by melap** [comment]
Staging link: http://apache-beam-website-pull-requests.storage.googleapis.com/394/documentation/io/built-in/index.html

File: src/documentation/io/built-in.md [hunk 1]
```diff
@@ -24,11 +24,14 @@ Consult the [Programming Guide I/O section]({{site.baseurl }}/documentation/prog
 <tr>
   <td>Java</td>
   <td>
-    <p><a href="https://github.com/apache/beam/tree/master/sdks/java/io/hadoop-file-system">Apache Hadoop File System</a></p>
+    <p>Beam Java supports HDFS, S3, GCS and local filesystems out of the
```
Comment (melap) [inline src/documentation/io/built-in.md]: By "out of the box", is that saying "you don't need to specify a dependency"? Could we simplify it to:
Apache HDFS, Amazon S3, Google Cloud Storage, and local filesystems (no dependency needed)

I think this whole table is overdue for an overhaul -- perhaps as one IO per line that checks off what languages support it and lists the required dependency if applicable. I'll put that on my list.

File: src/documentation/io/built-in.md [hunk 2]
```diff
@@ -24,11 +24,14 @@ Consult the [Programming Guide I/O section]({{site.baseurl }}/documentation/prog
...
+    box.</p>
+    <p><a href="https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileIO.java">FileIO</a> (general-purpose reading, writing and matching of files)</p>
```
Comment (melap) [inline src/documentation/io/built-in.md]: add comma: writing,

File: src/documentation/io/built-in.md [hunk 3]
```diff
@@ -56,6 +60,7 @@ Consult the [Programming Guide I/O section]({{site.baseurl }}/documentation/prog
 <tr>
   <td>Python</td>
   <td>
+    <p>Beam Python supports GCS and local filesystems out of the box. Support for HDFS is in progress (<a href="https://issues.apache.org/jira/browse/BEAM-3099">BEAM-3099</a>).</p>
```
Comment (melap) [inline src/documentation/io/built-in.md]: Similarly, though not sure about the equivalent dependency statement for Python?
Google Cloud Storage and local filesystems

Comment (melap) [inline src/documentation/io/built-in.md]: Apache HDFS
Should this second part be added to the table below instead?

Comment (melap) [inline src/documentation/io/built-in.md]: Should we add Variant Call Format (VCF) (I think added in 2.3.0?) for Python?

File: src/documentation/io/built-in.md (hunk 1 above)
Comment (jkff) [reply src/documentation/io/built-in.md]: Done (but without "no dependency needed": I'm not aware of anyone developing a filesystem support for Beam for which a dependency is needed)

File: src/documentation/io/built-in.md (hunk 2 above)
Comment (jkff) [reply src/documentation/io/built-in.md]: Done

File: src/documentation/io/built-in.md (hunk 3 above)
Comment (jkff) [reply src/documentation/io/built-in.md]: Same.

Comment (jkff) [reply src/documentation/io/built-in.md]: Good point, moved.

Comment (jkff) [reply src/documentation/io/built-in.md]: Done.

**Comment by jkff** [comment]
Thanks, PTAL.

**Comment by melap** [comment]
@asfgit merge

**Pull request closed without merging**